File: src/features/presence/data/dto/presence.dto.ts
import type { PresenceType } from "@/features/presence/domain/entities/presence";

/** Wire shape returned by the Rust commands (serialized camelCase). */
export interface PresenceDto {
  id: string;
  profileId: string;
  day: number;
  type: PresenceType;
  co2Kg: number | null;
  isEstimated: boolean;
  createdAt: number;
  updatedAt: number;
}

/** A persisted trip snapshot returned by `get_presence_trips`. */
export interface PresenceTripDto {
  id: string;
  modeId: string;
  distanceKm: number;
  roundTrip: boolean;
  occupants: number;
  co2Kg: number;
  isEstimated: boolean;
  position: number;
}

File: src-tauri/src/application/dto/presence_dto.rs
use serde::Serialize;

use crate::domain::entities::presence::Presence;

/// Presence shape returned to the frontend (serialized camelCase). The `kind`
/// field is exposed on the wire as `type`.
#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct PresenceDto {
    pub id: String,
    pub profile_id: String,
    pub day: i64,
    #[serde(rename = "type")]
    pub kind: String,
    pub created_at: i64,
    pub updated_at: i64,
}

impl From<Presence> for PresenceDto {
    fn from(p: Presence) -> Self {
        Self {
            id: p.id,
            profile_id: p.profile_id,
            day: p.day,
            kind: p.kind.as_str().to_string(),
            created_at: p.created_at,
            updated_at: p.updated_at,
        }
    }
}
